app/modules/vm/index.coffee

```diff
@@ -472,11 +475,17 @@ module.exports = angular.module 'xoWebApp.vm', [
 
         return
 
+      # Disk resize
+      forEach disks, (attributes, id) ->
+        disk = get id
+        if attributes.size isnt disk.size
+          promises.push (xo.disk.resize id, attributes.size)
+
       forEach disks, (attributes, id) ->
         # Keep only changed attributes.
         disk = get id
         forEach attributes, (value, name) ->
-          delete attributes[name] if value is disk[name]
+          delete attributes[name] if value is disk[name] or name is 'size'
```

Comment:
size changes are handled with disk.resize just before, or did I miss the whole point of the issue ?...trying to let a human readable size here led me to a vdi.set crash, but perhaps it was before recent changes on server side ?...